### ISSUE TITLE
Doesn't work with hub 2.0

### ISSUE THREAD

**Issue by asimpson** [thoughtbot/gitsh#206]
Launching gitsh via: `gitsh --git $(which hub)` works for normal git functions but for additional hub functionality like `browse`, gitsh returns the standard git error: `git: 'browse' is not a git command.`

Using hub outside of gitsh works just fine, e.g. `hub browse` or `git browse` at the standard command line works as expected.

Any ideas how to get this working?

**Comment by georgebrock** [comment]
This seems to be because `hub` doesn't support the `-c` flag:

```
$ hub pull-request
Aborted: head branch is the same as base ("master")
(use `-h <branch>` to specify an explicit pull request head)
$ hub -c foo.bar=1 pull-request
git: 'pull-request' is not a git command. See 'git --help'.
```

gitsh relies heavily on the `-c` flag to pass session-specific settings from the environment to Git.

**Comment by georgebrock** [comment]
@asimpson It should be possible to work around this bug: if you set the `gitsh.gitCommand` setting in your `~/.gitconfig` file (e.g. run `git config --global gitsh.gitCommand $(which hub)`) instead of using the `--git` option, then gitsh won't need to pass a `-c` option to Git commands.

This will work as long as you don't use gitsh's Git configuration variables (i.e. you don't run commands like `:set user.name "A temporary name"` in gitsh). If you do use gitsh's Git configuration variables, it'll pass one or more `-c` options to each Git command, and `hub` won't work.

**Comment by asimpson** [comment]
@georgebrock thanks for the hard work on this! That worked out perfectly for me, thanks so much! I'll close this issue since you have the root problem logged in #756.

**Comment by georgebrock** [comment]
@asimpson No problem. Since the root issue is on a different repo, I'd prefer to leave this open so it'll be easier to find for other gitsh users. Hopefully we can close it again when the hub issues is fixed.

**Comment by georgebrock** [comment]
The underlying hub issue has been fixed (thanks @jingweno!) so we can close this one after the next hub release.

**Comment by lukerollans** [comment]
+1 thanks for temp workaround!

**Comment by rmeritz** [comment]
@georgebrock - Tack så mycket! For making it so easy to let me know how to fix this.

**Comment by georgebrock** [comment]
@rmeritz Varsågod!

**Comment by georgebrock** [comment]
Looks like the fix was released in Hub 2.2.0: https://github.com/github/hub/releases/tag/v2.2.0

Closing.
